src/Neo/Wallets/Helper.cs

```diff
@@ -126,35 +128,76 @@ public static long CalculateNetworkFee(this Transaction tx, DataCache snapshot,
                     if (md.ReturnType != ContractParameterType.Boolean)
                         throw new ArgumentException("The verify method doesn't return boolean value.");
                     if (md.Parameters.Length > 0 && invocationScript is null)
-                        throw new ArgumentException("The verify method requires parameters that need to be passed via the witness' invocation script.");
+                    {
+                        var script = new ScriptBuilder();
+                        foreach (var par in md.Parameters)
+                        {
+                            switch (par.Type)
+                            {
+                                case ContractParameterType.Any:
+                                case ContractParameterType.Signature:
+                                case ContractParameterType.String:
+                                case ContractParameterType.ByteArray:
+                                    script.EmitPush(new byte[64]);
+                                    break;
+                                case ContractParameterType.Boolean:
+                                    script.EmitPush(true);
+                                    break;
+                                case ContractParameterType.Integer:
+                                    script.Emit(OpCode.PUSHINT256, new byte[Integer.MaxSize]);
+                                    break;
+                                case ContractParameterType.Hash160:
+                                    script.EmitPush(new byte[UInt160.Length]);
+                                    break;
+                                case ContractParameterType.Hash256:
+                                    script.EmitPush(new byte[UInt256.Length]);
+                                    break;
+                                case ContractParameterType.PublicKey:
+                                    script.EmitPush(new byte[33]);
+                                    break;
+                                case ContractParameterType.Array:
+                                    script.Emit(OpCode.PUSHINT256, new byte[Integer.MaxSize]);
+                                    script.Emit(OpCode.NEWARRAY);
+                                    break;
+                                case ContractParameterType.Map:
+                                    script.Emit(OpCode.NEWMAP);
+                                    break;
+                                default:
+                                    throw new ArgumentException("The verify method requires parameters that need to be passed via the witness' invocation script.");
```

Comment:
It differs slightly from NeoGo behaviour, we need to discuss it: NeoGo doesn't return error on unknown type; instead it tries to do the best and emit as much parameters as possible, skipping the others. Then "verify" method is processed in any case, and an error is returned only if "verify" doesn't return boolean. I think that this approach is good because it tries to do as much as possible without extra user's data.

Reply:
Yes you need to support `contract` as signer with `verify` function.

Reply:
The only thing that is missing is Map and InteropInterface currently. For maps some zero value can be added (empty map), but for interop interfaces it's harder to say, maybe NULL.

Reply:
> The only thing that is missing is Map and InteropInterface currently. For maps some zero value can be added (empty map), but for interop interfaces it's harder to say, maybe NULL.

https://github.com/neo-project/neo/pull/3385#discussion_r1665156066

Reply:
OK, but then throwing an error is not the best way to handle this.

Reply:
@roman-khimov done